commands/execute.py

```
@@ -147,97 +146,599 @@ def get_image_filename(i):
+    def get_optimal_worker_count():
+        import multiprocessing
+        cores = multiprocessing.cpu_count()
+        return max(1, min(6, int(cores * 0.75)))
+    
+    def use_videotoolbox_encoding():
+        if not is_apple_silicon():
+            return False
+        try:
+            result = subprocess.run(["ffmpeg", "-encoders"], stdout=subprocess.PIPE, text=True, check=False)
+            return "h264_videotoolbox" in result.stdout
+        except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)